services/rds/src/main/java/org/finra/gatekeeper/services/auth/GatekeeperRoleService.java

```diff
@@ -181,4 +225,64 @@ private GatekeeperRdsRole checkGatekeeperRdsRole() {
             return GatekeeperRdsRole.UNAUTHORIZED;
         }
     }
+
+
+
+    public Map<String, Map<String, Map<String, Integer>>> getApprovalPolicy(Map<String, RoleMembership> roleMemberships) {
+        Map<String, Map<String, Map<String, Integer>>> approvalPolicy = new HashMap<>();
+        for(String application : roleMemberships.keySet()) {
+            approvalPolicy.put(application, getApprovalPolicyByApplication(roleMemberships.get(application)));
+        }
+        return approvalPolicy;
+    }
+
+    private Map<String, Map<String, Integer>> getApprovalPolicyByApplication(RoleMembership roleMembership) {
+        gatekeeperRoles = new ArrayList<>();
+        gatekeeperRoles.add("datafix");
+        gatekeeperRoles.add("readonly_confidential");
+        gatekeeperRoles.add("readonly");
+        gatekeeperRoles.add("dba_confidential");
+        gatekeeperRoles.add("dba");
+
+        sdlcs = new ArrayList<>();
+        sdlcs.add("dev");
+        sdlcs.add("qa");
+        sdlcs.add("prod");
+
+        Map<String, Map<String, Integer>> applicationApprovalPolicy = initializeApprovalPolicy();
+
+        for(GatekeeperRdsRole role : roleMembership.getRoles().keySet()) {
+            Map<String, Map<String, Integer>> roleApprovalPolicyToCompareTo = gatekeeperApprovalProperties.getApprovalPolicy(role);
+            applicationApprovalPolicy = mergeApprovalPolicies(applicationApprovalPolicy, roleApprovalPolicyToCompareTo);
+        }
+
+
+        return applicationApprovalPolicy;
+    }
+
+    private Map<String, Map<String, Integer>> initializeApprovalPolicy() {
+        Map<String, Map<String, Integer>> initialApprovalPolicy = new HashMap<>();
+
+        for(String gatekeeperRole : gatekeeperRoles) {
+            initialApprovalPolicy.put(gatekeeperRole, new HashMap<>());
```

Comment:
You should just initialize the hashMap first, put all the values in it and then throw it into initialApprovalPolicy